front-end-tools/src/components/ReadComponent.tsx

```diff
@@ -96,14 +96,12 @@ export default function ReadComponenet(props: ConnectionProps) {
                 throw new Error('Set smart contract name');
             }
 
-            let schema = '';
-
-            const schemaFromModule = deriveContractInfo ? embeddedModuleSchemaBase64 : uploadedModuleSchemaBase64;
-
-            if (schemaFromModule !== undefined) {
-                schema = schemaFromModule;
+            const schema = deriveContractInfo ? embeddedModuleSchemaBase64 : uploadedModuleSchemaBase64;
+            if (schema === undefined) {
+                setSchemaError('Schema was not uploaded');
```

Comment:
If a first-time user does the following thing in the read box:
- inserts the contract `6956 (no embedded schema)`
- checks the checkbox `derive from smart contract index`
- selects an entry point name
- checks the checkbox `has input parameter`

Now the error occurs: `Schema was not uploaded`
Since there is nothing in the read box where the user could upload the schema, it is confusing.

Suggestion: Use the following error message instead which has additional information on how to resolve the above scenario.

`Schema was not uploaded or not embedded into the module. Uncheck "Derive From Smart Contract Index" checkbox to manually upload a schema or uncheck "Has Input Paramter" checkbox if this entrypoint has no input parameter. `

Reply:
It seems to me that the "has input parameter" box shouldn't be there when deriving from the contract, but I don't know how difficult it would be to change.

Reply:
Good catch!